test/functional/change_stream.test.js

```diff
@@ -2597,233 +2599,236 @@ describe('Change Streams', function() {
-          })(),
-          getMore: (function*() {
-            yield { numDocuments: 1, postBatchResumeToken: false };
-          })()
         });
-        let token;
+      }
+    });
 
-        return manager
-          .ready()
-          .then(() => {
-            const changeStream = manager.makeChangeStream();
-            let counter = 0;
-            changeStream.cursor.on('response', () => {
-              if (counter === 1) {
-                token = changeStream.resumeToken;
-              }
-              counter += 1;
+    // 18. $changeStream stage for ChangeStream started with startAfter against a server >=4.1.1
+    // that has received at least one result
+    // - MUST include a resumeAfter option
+    // - MUST NOT include a startAfter option when resuming a change stream.
+    it('$changeStream that has received results must include resumeAfter and not startAfter', {
+      metadata: { requires: { topology: 'replicaset', mongodb: '>=4.1.1' } },
+      test: function(done) {
+        const changeStream = coll.watch([], { startAfter });
+        tryNext(changeStream, (err, change) => {
+          expect(err).to.not.exist;
+          expect(change).to.containSubset({
+            operationType: 'insert',
+            fullDocument: { x: 2 }
+          });
+          coll.insertOne({ x: 3 }, { w: 'majority', j: true }, err => {
+            expect(err).to.not.exist;
+            triggerResumableError(changeStream);
+            tryNext(changeStream, (err, change) => {
+              expect(err).to.exist;
+              expect(change).to.not.exist;
+              const retry = setInterval(
+                () =>
+                  tryNext(changeStream, (err, change) => {
+                    if (err) {
+                      return;
+                    }
+                    clearInterval(retry);
+                    expect(change).to.containSubset({
+                      operationType: 'insert',
+                      fullDocument: { x: 3 }
+                    });
+                    const pipeline = changeStream.cursor.operation.pipeline;
+                    expect(pipeline).to.be.an('array');
+                    expect(pipeline[0]).to.have.property('$changeStream');
+                    expect(pipeline[0].$changeStream).to.have.property('resumeAfter');
+                    expect(pipeline[0].$changeStream).to.not.have.property('startAfter');
+                    changeStream.close(done);
+                  }),
+                1000
+              );
             });
-
-            // Note: this is expected to fail
-            return changeStream.next();
-          })
-          .then(
-            () => manager.teardown(),
-            err => manager.teardown(err)
-          )
-          .then(() => {
-            expect(token).to.not.exist;
           });
-      });
+        });
+      }
     });
   });
 });
+
+describe('Change Stream Resume Error Tests', function() {
```

Comment:
this looks like it belongs inside the `describe("ChangeStreams"...` block above, maybe even something like:
```
describe('ChangeStream', function() {
  context('isResumableError', function() {
    it('should do something', function() {});
  });
});
```